src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -823,9 +866,13 @@ bool emberAfOperationalCredentialsClusterAddTrustedRootCertificateCallback(
 
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: commissioner has added a trusted root Cert");
 
-    VerifyOrExit(Server::GetInstance().GetCommissioningWindowManager().CommissioningWindowStatus() !=
-                     AdministratorCommissioning::CommissioningWindowStatus::kWindowNotOpen,
-                 status = EMBER_ZCL_STATUS_FAILURE);
+    DeviceControlServer * server = &DeviceLayer::DeviceControlServer::DeviceControlSvr();
+
+    if (!server->IsFailSafeArmed())
+    {
+        LogErrorOnFailure(commandObj->AddStatus(commandPath, Protocols::InteractionModel::Status::UnsupportedAccess));
+        return true;
+    }
 
     // TODO: Ensure we do not duplicate roots in storage, and detect "same key, different cert" errors
```

Comment:
Fwiw, filed https://github.com/project-chip/connectedhomeip/issues/15311 on this having some issues.